app/Http/Controllers/WatchlistController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use App\Models\Movie;
+use App\Models\MovieUser;
+use Illuminate\Http\Request;
+
+class WatchlistController extends Controller
+{
+    public function index(User $user)
+    {
+        if(auth()->user()->id == $user->id) {
+            $watchlist = $user->movies()
+                ->select('id', 'title', 'poster')
+                ->orderByDesc('movie_user.created_at')
+                ->paginate(15);
+
+            return view('watchlist/index',[
+                'watchlist' => $watchlist,
+                'user' => $user
+            ]);
+        } else {
+            return redirect('/')
+                ->with('message', 'You don\'t have access to that page');
+        }
+    }
+
+    public function create(Request $request, User $user)
+    {
+        if(auth()->user()->id == $user->id) {
+            $keyword = $request->input('search');
+
+            $movies = Movie::query()
+                ->select('id', 'title', 'poster')
+                ->when($keyword, function ($q) use ($keyword){
+                    $q->where('title', 'LIKE', "%{$keyword}%");
+                })
+                ->orderBy('title')
+                ->paginate(10);
+
+
+            return view('watchlist/create', [
+                'keyword' => $request->input('title'),
+                'movies' => $movies,
+                'user' => $user
+            ]);
+        } else {
+            return redirect('/')
+                ->with('message', 'You don\'t have access to that page');
+        }
+    }
+
+    public function store(Request $request, User $user)
+    {
+        if(auth()->user()->id == $user->id) {
+            if ($user->movies()->where('id',$request->input('movie_id'))->exists()) {
+                return redirect("profile/$user->id/watchlist")
+                    ->with('message', 'Movie not added, already on your watchlist');
+            } else {
+                $user->movies()
+                    ->syncWithoutDetaching([$request->input('movie_id')]);
+
+                return redirect("profile/$user->id/watchlist")
+                    ->with('message', 'Movie successfully added to your watchlist');
+            }
+        } else {
+            return redirect('/')
```

Comment:
this is a check for guest right?

I don't think you need this as you are already using the auth middleware.

You can instead do 
```
if (auth()->user()->id !== $user->id ) { return redirect} 

if($user->movies()->where('id',$request->input('movie_id'))->exists()) { return redirect } 

 $user->movies()->syncWithoutDetaching([$request->input('movie_id')]);
                return redirect("profile/$user->id/watchlist")
                    ->with('message', 'Movie successfully added to your watchlist');
```